operator/pkg/awsprovider/instances/reconciler.go

```diff
@@ -180,24 +180,36 @@ func (c *Controller) subnetsFor(ctx context.Context, dataplane *v1alpha1.DataPla
 	if err != nil {
 		return nil, fmt.Errorf("getting subnet for %s, %w", dataplane.Spec.ClusterName, err)
 	}
-	return c.filterPrivateSubnets(ctx, subnetIDs)
+	// Returns public subnets if no private subnets are found
+	return c.filterSubnets(ctx, subnetIDs)
 }
 
-func (c *Controller) filterPrivateSubnets(ctx context.Context, ids []*string) ([]string, error) {
+func (c *Controller) filterSubnets(ctx context.Context, ids []*string) ([]string, error) {
+	if len(ids) == 0 {
```

Comment:
It looks like theres some redundancy with the emptiness checking logic. 

Previously this filter function would simply return [] if no subnets were found, rather than erroring. [Callers](https://github.com/awslabs/kubernetes-iteration-toolkit/pull/233/files#diff-295f7552e000471288f85f68537af958d7074254753d9a3935a85bde73db8ac9R124) were responsible for behaving correctly if there weren't any subnets. 

It's important to differentiate between transient network errors (i.e. retriable) and permanent logical errors (e.g. no subnets configured). If you comingle these concepts, you risk your code failing rarely for unknown reasons.

Reply:
Changed to the empty slice return per our conversation on Friday.